lib/topics_filter.rb

```diff
@@ -272,10 +273,15 @@ def self.option_info(guardian)
             name: "in:watching_first_post",
             description: I18n.t("filter.description.in_watching_first_post"),
           },
+          { name: "in:new", description: I18n.t("filter.description.in_new") },
+          { name: "in:new-posts", description: I18n.t("filter.description.in_new_posts") },
+          { name: "in:new-topics", description: I18n.t("filter.description.in_new_topics") },
         ],
       )
     end
 
+    DiscoursePluginRegistry.apply_modifier(:topics_filter_options, results, guardian)
+
     if SiteSetting.tagging_enabled?
```

Comment:
Maybe add an indicator that this relates to the UI? 

Also, a minor 🏌️, could we move this modifier to line 309 (just before returning `results`) so that any plugin that attaches to this modifier has all the results before modifying it?